tests/integrations/test_db_integrations_sql.py

```diff
@@ -192,9 +210,26 @@ def test_error_stack(self):
         span = self._get_sql_span_for_request(self.requests[self.db_service]["select_error"])
         assert span["meta"]["error.stack"].strip()
 
+    def _get_sql_span_for_request(self, weblog_request):
+        """To be implemented by subclasses"""
+        return {}
+
+
+@scenarios.integrations
+class _BaseTracerIntegrationsSqlTestClass(_BaseIntegrationsSqlTestClass):
+    @missing_feature(
+        library="java",
+        reason="The Java tracer normalizing the SQL by replacing literals to reduce resource-name cardinality",
+    )
+    def test_NOT_obfuscate_query(self):
+        """ All queries come out without obfuscation from tracer library """
+        for db_operation, request in self.requests[self.db_service].items():
+            span = self._get_sql_span_for_request(request)
+            assert span["resource"].count("?") == 0
```

Comment:
Maybe a assertion comment explaining the why . And also, `"?" not in span["resource"]` is probably easier to read for non python engineer.

Reply:
done!